src/solidity-frontend/solidity_grammar.cpp

```diff
@@ -225,13 +225,16 @@ TypeNameT get_type_name_t(const nlohmann::json &type_name)
       // FunctionToPointer decay in CallExpr when making a function call
       return Pointer;
     }
-    else if (
-      type_name["typeIdentifier"].get<std::string>().find("ArrayToPtr") !=
-      std::string::npos)
+    else if (typeIdentifier.find("ArrayToPtr") != std::string::npos)
     {
       // ArrayToPointer decay in DeclRefExpr when dereferencing an array, e.g. a[0]
       return PointerArrayToPtr;
     }
+    // for Special Variables and Functions
+    else if (typeIdentifier.substr(0, 7) == "t_magic")
```

Comment:
Shall we use the `std::string::compare` function instead of `substr` for a more efficient comparison?